hack/generator/pkg/codegen/pipeline_augment_status.go

```diff
@@ -70,15 +71,18 @@ func augmentResourcesWithStatus(idFactory astmodel.IdentifierFactory, config *co
 			for typeName, typeDef := range types {
 				// for resources, try to find the matching Status type
 				if resource, ok := typeDef.Type().(*astmodel.ResourceType); ok {
+					var newStatus astmodel.Type
 					if statusDef, ok := statusTypes.resourceTypes.tryFind(typeName); ok {
 						klog.V(4).Infof("Swagger information found for %v", typeName)
-						newTypes.Add(astmodel.MakeTypeDefinition(typeName, resource.WithStatus(statusDef)))
+						newStatus = statusDef
 						found++
 					} else {
-						// missing status is caught later in pipeline (checkForMissingStatusInformation)
-						// as we only want to report this for non-pruned types
-						newTypes.Add(typeDef)
+						klog.V(4).Infof("Swagger information missing for %v", typeName)
+						// add a warning that the status is missing
+						// this will be reported if the type is not pruned
+						newStatus = astmodel.MakeErroredType(nil, nil, []string{fmt.Sprintf("missing status information for %v", typeName)})
```

Comment:
Why not just put the error on the resource rather than allowing ErroredType to wrap nil? Is it because we want to allow generation of the resource even if the status is missing? Although I guess that's what warnings are for. Some other reason?

Reply:
Wrapping the ResourceType would introduce a lot more churn/changes since we pattern-match on ResourceType in many places. (part of the problem with having wrapper types instead of some kind of annotation/attribute)

Reply:
Ah right - makes sense, thanks.